lib/services/security/permissions-tree.js

```diff
@@ -15,10 +15,15 @@ module.exports = class PermissionsTree {
 
   mergeBranch(permissions) {
     return (tree, path) =>
-      _.set(tree, path.split('/').slice(path.startsWith('/') ? 1 : 0), {
-        $leaf: path,
-        ...permissions[path]
-      });
+      _.setWith(
```

Comment:
why setWith instead of set?

Reply:
Because if you don't specify the setWith(...., Object), on a numeric value it creates an Array instead of an object.